src/OpenTelemetry.Extensions.Tracing/OpenTelemetry.Extensions.Tracing.csproj

```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <!-- OmniSharp/VS Code requires TargetFrameworks to be in descending order for IntelliSense and analysis. -->
+    <TargetFrameworks>netstandard2.1;netstandard2.0</TargetFrameworks>
+    <Description>Extensions for using OpenTelemetry with System.Diagnostics.Tracing</Description>
```

Comment:
I know the name `OpenTelemetry.Extensions.Tracing` is going to be a lightning rod but I'll explain my thinking here.

My intention was this project would contain extensions for things in the `System.Diagnostics.Tracing` namespace. `EventSource` but also [EventCounters](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.EventCounters). IMO OpenTelemetry.Instrumentation.EventCounters is misnamed.

Anyway open to alternatives just thought I would explain myself 😄

Reply:
> My intention was this project would contain extensions for things in the System.Diagnostics.Tracing namespace.

Then we could probably add `Diagnostics` in the package name: `OpenTelemetry.Extensions.Diagnostics.Tracing` because just `OpenTelemetry.Extensions.Tracing` might suggest that these would be extensions for OTel Traces covering things such as [AutoFlushActivityProcessor ](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Extensions/Trace/AutoFlushActivityProcessor.cs)

Reply:
⚡ ⚡ !! Haha, yea naming is a bummer. We should all just stop using words.

Understood that `System.Diagnostics.Tracing` namespace involves more than `EventSource`, but were you envisioning that this package might one day involve itself with `EventCounters` somehow?

`OpenTelemetry.Extensions.EventSource` was my favorite of my suggestions here
https://github.com/open-telemetry/opentelemetry-dotnet/pull/3305#discussion_r917144381. But I could get behind the `OpenTelemetry.Extensions.Diagnostics.Tracing` or even more precisely including the whole namespace `OpenTelemetry.Extensions.System.Diagnostics.Tracing` ... even though it gives me a minor headache 🥴.

Reply:
Could we have separate packages for `OpenTelemetry.Extensions.EventCounter` and `OpenTelemetry.Extensions.EventSource`? (e.g. I guess there are far more developers who need EventSource integration comparing to EventCounter)

I prefer separate packages because they can be deprecated independently.

Reply:
OK I'll rename this `OpenTelemetry.Extensions.EventSource` and get to work on tests. @utpilla You want me to open issue to rename `OpenTelemetry.Instrumentation.EventCounters` -> `OpenTelemetry.Extensions.EventCounter`?

Reply:
Oh yea, you've reminded me of the other thing that was on my mind here https://github.com/open-telemetry/opentelemetry-dotnet/pull/3438#discussion_r919496560.

What do we consider an `Extension` vs. `Instrumentation`? Something about the support you're adding for various logging frameworks _feels_ like an extension to me... but I can't really articulate why they feel less like instrumentation. Do other folks have this feeling? If so, can you articulate it better than me?

Reply:
In my mind...

* If a library is already emitting signals and we are translating that into OTel, that is an extension.

* If a library doesn't emit signals at all and we are hooking into it to emit OTel, that is instrumentation.

🤷

Reply:
Instrumentation Library - the component which emits signals using OTel API. (it could be emitting for its own, or on behalf of an instrumented library)

Extension - Does not call OTel API, but pipes data through OTEL SDK.. (so the logging stuffs are extensions, as they do not call the Logging API)

If we write a component for HttpClient, which in turn calls iLogger.Log API, then I'd call it an InstrumentationLibrary for HttpClient.

If we write a component for piping EventSource/Serilog data to OTel SDK, without calling ILogger.Log API, then I'd call it an Extension..

👋

Reply:
> @utpilla You want me to open issue to rename OpenTelemetry.Instrumentation.EventCounters -> OpenTelemetry.Extensions.EventCounter?

I agree with @cijothomas 's definition of an instrumentation library. By that definition, the [EventCounters ](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.EventCounters)library is an instrumentation library and we don't have to rename it to Extensions.

Reply:
I think we're getting closer to a shared mental model, but I'm not entirely sold yet 😄. I'd like to open an issue to continue this conversation cause I think it's a good one and will make future naming of things easier.

I think for this PR it sounds like we're in agreement that `OpenTelemetry.Extensions.EventSource` is an adequate name. Let's not touch anything else just yet.

Reply:
Hmm interesting. Hard time wrapping my mind around this. Is this definition from the spec somewhere? User plugging this stuff in is doing so at the host level so I'm not sure if the API/SDK distinction is useful.

Just eyeballing this list...

![image](https://user-images.githubusercontent.com/28367120/180298025-9de5843b-840a-45f2-8c36-742e5e1c3b9c.png)

Which thing is not like the others? 🍎 🍎 🍎 🍌 🤣

OpenTelemetry.Instrumentation.AspNet - Emits telemetry about AspNet
OpenTelemetry.Instrumentation.MassTransit - Emits telemetry about MassTransit
OpenTelemetry.Instrumentation.Runtime - Emits telemetry about.NET runtime
OpenTelemetry.Instrumentation.StackExchangeRedis - Emits telemetry about Redis
...etc...
OpenTelemetry.Instrumentation.EventCounters - Transposes telemetry written about _whatever_ using EventCounters into System.Diagnostics.DiagnosticSource telemetry. Does not emit telemetry _about_ EventCounters

IMO the HOW it is done is less important than WHAT it is doing.

Reply:
>  Is this definition from the spec somewhere?

No. My view !
Only spec defined term is https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#instrumentation-library

We need to definitely discuss this more.

Reply:
> IMO the HOW it is done is less important than WHAT it is doing.

💯

Reply:
Yes 💯 to that as well. I think I've captured the spirit of `WHAT` in the issue I just opened #3469.

Also, per my thoughts in #3469, I've changed my mind and I think this package should be `OpenTelemetry.Shims.EventSource` 😆.

Reply:
> Which thing is not like the others? 🍎 🍎 🍎 🍌 🤣

Also, rename `OpenTelemetry.Extensions.EventCounter` -> `OpenTelemetry.🍌`